src/app/features/tech-record/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.ts

```diff
@@ -1,38 +1,48 @@
-import { Component, ViewChild } from '@angular/core';
+import { Component, OnDestroy, ViewChild } from '@angular/core';
 import { FormGroup } from '@angular/forms';
 import { ActivatedRoute, Router } from '@angular/router';
 import { GlobalErrorService } from '@core/components/global-error/global-error.service';
 import { VehicleTechRecordModel, VehicleTypes } from '@models/vehicle-tech-record.model';
 import { Store } from '@ngrx/store';
 import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
-import { createVehicleRecord } from '@store/technical-records';
+import { createVehicleRecord, createVehicleRecordSuccess } from '@store/technical-records';
 import { TechnicalRecordServiceState } from '@store/technical-records/reducers/technical-record-service.reducer';
-import { map, Observable, take, tap, withLatestFrom } from 'rxjs';
+import { map, Observable, Subject, take, takeUntil, withLatestFrom } from 'rxjs';
 import { TechRecordSummaryComponent } from '../../../components/tech-record-summary/tech-record-summary.component';
+import { Actions, ofType } from '@ngrx/effects';
 
 @Component({
   selector: 'app-hydrate-new-vehicle-record',
   templateUrl: './hydrate-new-vehicle-record.component.html'
 })
-export class HydrateNewVehicleRecordComponent {
+export class HydrateNewVehicleRecordComponent implements OnDestroy {
   @ViewChild(TechRecordSummaryComponent) summary?: TechRecordSummaryComponent;
   isInvalid: boolean = false;
   batchForm?: FormGroup;
 
+  private destroy$ = new Subject<void>();
+
   constructor(
+    private actions$: Actions,
     private globalErrorService: GlobalErrorService,
     private route: ActivatedRoute,
     private router: Router,
     private store: Store<TechnicalRecordServiceState>,
     private technicalRecordService: TechnicalRecordService
-  ) {}
+  ) {
+    this.actions$.pipe(ofType(createVehicleRecordSuccess), take(1), takeUntil(this.destroy$)).subscribe(({ vehicleTechRecords }) => {
```

Comment:
can't have both `take(1)` and `takeUntil()`

Reply:
You're right, I've removed the `take(1)`.